clients/java/src/main/java/io/camunda/zeebe/client/impl/worker/JobWorkerImpl.java

```diff
@@ -207,19 +214,30 @@ private void onPollError(final JobPoller jobPoller, final Throwable error) {
 
   /** Apply the backoff strategy by scheduling the next poll at a new interval */
   private void backoff(final JobPoller jobPoller, final Throwable error) {
+    getPollInterval();
+    LOG.debug(
+        "Failed to activate jobs due to {}, delay retry for {} ms",
+        error.getMessage(),
+        pollInterval);
+
+    releaseJobPoller(jobPoller);
+    schedulePoll();
+  }
+
+  private void backoffOnSuccessfulPollAndEmptyResponse(final JobPoller jobPoller) {
```

Comment:
:x: I think we can improve the deduplication here :slightly_smiling_face: 

So the common behavior between both methods is:

```
getPollInterval();
releaseJobPoller(jobPoller);
schedulePoll();
```

The only difference is that the error case will log something. So in a way, it's extending the common behavior. So we could instead do:

```java
  private void onPollError(final JobPoller jobPoller, final Throwable error) {
    backoff(jobPoller);
    LOG.debug(
      "Failed to activate jobs due to {}, delay retry for {} ms",
      error.getMessage(),
      pollInterval);
  }

  private void backoff(final JobPoller jobPoller) {
    getPollInterval();
    releaseJobPoller(jobPoller);
    schedulePoll();
  }
```

This works because `backoff` has already updated the `pollInterval`, so we can simply reuse it in the log message after calling `backoff` in `onPollError`. It's a little dirty, but I think it's fine as is.

I would also suggest adding a trace statement in `onPollSuccess` as well, after the `backoff` call to also log why we're backing off, e.g.

```java
    if (jobStreamer.isOpen() && activatedJobs == 0) {
      // to keep polling requests to a minimum, if streaming is enabled, and the response is empty,
      // we back off on poll success responses.
      backoff(jobPoller);
      LOG.trace("No jobs to activate via polling, will backoff and poll in {}", pollInterval);
    } else {
```